app/models/manageiq/providers/nuage/network_manager/cloud_subnet.rb

```diff
@@ -1,2 +1,7 @@
 class ManageIQ::Providers::Nuage::NetworkManager::CloudSubnet < ::CloudSubnet
+  before_destroy :remove_network_ports, prepend: true
+
+  def remove_network_ports
```

Comment:
@Ladas so this manual deletion is needed because Rails won't remove actual NetworkPorts based on `:dependent => :destroy` but only the records in intermediate table.

Reply:
hm, we usually delete the network ports as part of the Vms

can you describe a detail usecase/workflow for this to happen?